### PR TITLE
feat(backend): delete notification when user unffollow current user tm-608

### PR DESCRIPTION
**Pull request by Vfivew**
…llow another user tm-608

### CODE REVIEW

File: apps/backend/src/modules/notifications/notification.repository.ts [hunk 1]
```diff
@@ -73,6 +73,21 @@ class NotificationRepository implements Repository<NotificationEntity> {
...
+		id: number,
+		userId: number,
+		type: string,
+	): Promise<boolean> {
+		const deletedItemsCount = await this.notificationModel
+			.query()
+			.delete()
+			.where("receiverUserId", userId)
+			.andWhere("userId", id)
+			.andWhere("type", type);
```
Comment (ShadowOfTheSpace) [inline apps/backend/src/modules/notifications/notification.repository.ts:86]:
```suggestion
      .where({ receiverUserId, type, userId: id })
```

**Review by ShadowOfTheSpace — CHANGES_REQUESTED** [review]

File: apps/backend/src/modules/notifications/notification.repository.ts (hunk 1 above)
Comment (what1s1ove) [reply apps/backend/src/modules/notifications/notification.repository.ts:86]: agreed

**Review by what1s1ove — CHANGES_REQUESTED** [review]

File: apps/backend/src/modules/friends/friend.service.ts [hunk 2]
```diff
@@ -103,6 +103,12 @@ class FriendService {
 			});
 		}
 
+		await this.notificationService.deteleAllNotificationByUserId(
```
Comment (fshabanov) [inline apps/backend/src/modules/friends/friend.service.ts:106]:
```suggestion
		await this.notificationService.deleteAllNotificationsByUserId(
```

File: apps/backend/src/modules/notifications/notification.service.ts [hunk 3]
```diff
@@ -105,6 +105,18 @@ class NotificationService implements Service {
 		return await this.notificationRepository.deleteByActionId(actionId, type);
 	}
 
+	public async deteleAllNotificationByUserId(
```
Comment (fshabanov) [inline apps/backend/src/modules/notifications/notification.service.ts:108]:
```suggestion
	public async deleteAllNotificationsByUserId(
```

**Review by fshabanov — CHANGES_REQUESTED** [review]

**Review by YourTeaChan — APPROVED** [review]

File: apps/backend/src/modules/notifications/notification.repository.ts [hunk 4]
```diff
@@ -73,6 +73,19 @@ class NotificationRepository implements Repository<NotificationEntity> {
 		return Boolean(deletedRowsCount);
 	}
 
+	public async deleteAllNotificationByParameters(
```
Comment (Oleksandra-Masiuk) [inline apps/backend/src/modules/notifications/notification.repository.ts:76]:
```suggestion
	public async deleteAllNotificationsByParameters(
```

File: apps/backend/src/modules/notifications/notification.service.ts [hunk 5]
```diff
@@ -77,6 +77,18 @@ class NotificationService implements Service {
 		return await this.notificationRepository.delete(notificationId);
 	}
 
+	public async deleteAllNotificationsByUserId(
+		id: number,
+		userId: number,
+		type: string,
+	): Promise<boolean> {
+		return await this.notificationRepository.deleteAllNotificationByParameters(
```
Comment (fshabanov) [inline apps/backend/src/modules/notifications/notification.service.ts:85]:
```suggestion
		return await this.notificationRepository.deleteAllNotificationsByParameters(
```
Now in some places you have Notification, in some places Notifications

**Review by fshabanov — CHANGES_REQUESTED** [review]

**Review by Oleksandra-Masiuk — CHANGES_REQUESTED** [review]

**Review by fshabanov — APPROVED** [review]

**Review by Ann-Maksay — APPROVED** [review]

**Review by what1s1ove — APPROVED** [review]

**Pull request merged**